src/chelonia.ts

```diff
@@ -1498,16 +1508,7 @@ export default sbp('sbp/selectors/register', {
     let reader: ReadableStreamDefaultReader<string>
     const s = new ReadableStream({
       start: async (controller) => {
-        const first = (await this.config
-          .fetch(`${this.config.connectionURL}/file/${startHash}`, {
-            signal: this.abortController.signal
-          })
-          .then(handleFetchResult('text'))) as unknown as string
-        const deserializedHEAD = SPMessage.deserializeHEAD(first)
-        if (deserializedHEAD.contractID !== contractID) {
-          controller.error(new Error('chelonia/out/eventsBetween: Mismatched contract ID'))
-          return
-        }
+        const deserializedHEAD = await sbp('chelonia/out/deserializedHEAD', startHash, { contractID })
```

Comment:
Somewhat confusing API imo.

1. The `contractID` appears to be optional as it's typed `contractID?: string`, yet the implementation always does the comparison, indicating it's mandatory.
2. So if it's mandatory I wouldn't split the parameters like this, but instead move `hash` into the object. Also better either way as it's named parameters like that and named parameters are easier to read at the call site